scripts/tools/zap_regen_all.py

```diff
@@ -38,6 +52,9 @@ def __init__(self, zap_config, template=None, output_dir=None):
         else:
             self.output_dir = None
 
+    def distinct_output(self):
+        return ZapDistinctOutput(input_template=self.template, output_directory=self.output_dir)
```

Comment:
Why does the input template matter?  I'd think that multiple things using the same output directory is bad no matter what... Or do we actually have multiple different input templates that populate different sets of files in the same output directory or something?

Reply:
My thought was that template contains the names of the outputs, so different templates may generate different file names, so even if the output directory is the same, the names may not collide.

I agree that same output directory seems wrong, but I am in "patching" mode here.

Reply:
It seems required because placeholder apps do this:

```python
            targets.append(ZAPGenerateTarget(filepath, output_dir=output_dir))
            targets.append(
                ZAPGenerateTarget(filepath, output_dir=output_dir, template=template))
            continue
```

Maybe that is fixable to, but that is a LOT of yak shaving.